sphinx/application.py

```diff
@@ -436,20 +436,29 @@ def disconnect(self, listener_id: int) -> None:
         logger.debug('[app] disconnecting event: [id=%s]', listener_id)
         self.events.disconnect(listener_id)
 
-    def emit(self, event: str, *args: Any) -> List:
+    def emit(self, event: str, *args: Any,
+             allowed_exceptions: Tuple["Type[Exception]", ...] = ()) -> List:
         """Emit *event* and pass *arguments* to the callback functions.
 
         Return the return values of all callbacks as a list.  Do not emit core
         Sphinx events in extensions!
+
+        .. versionchanged:: 3.1
+
+           Added *allowed_exceptions* to specify path-through exceptions
         """
         return self.events.emit(event, *args)
```

Comment:
I guess the ``allowed_exceptions`` should be passed on :-).

Reply:
oh, thanks!